src/main/java/net/sf/jabref/importer/fileformat/MedlinePlainImporter.java

```diff
@@ -123,145 +125,110 @@ public ParserResult importDatabase(BufferedReader reader) throws IOException {
-                    } else if ("".equals(type)) {
-                        type = "other";
-                    }
+                String label = entry.substring(0, entry.indexOf('-')).trim();
+                String value = entry.substring(entry.indexOf('-') + 1).trim();
 
-                } else if ("TI".equals(lab)) {
-                    String oldVal = hm.get("title");
-                    if (oldVal == null) {
-                        hm.put("title", val);
-                    } else {
-                        if (oldVal.endsWith(":") || oldVal.endsWith(".") || oldVal.endsWith("?")) {
-                            hm.put("title", oldVal + " " + val);
-                        } else {
-                            hm.put("title", oldVal + ": " + val);
-                        }
-                    }
+                if ("PT".equals(label)) {
+                    type = addSourceType(value, type);
                 }
-                // =
-                // val;
-                else if ("BTI".equals(lab) || "CTI".equals(lab)) {
-                    hm.put("booktitle", val);
-                } else if ("FAU".equals(lab)) {
+                addDates(hm, label, value);
+                addAbstract(hm, label, value);
+                addTitles(hm, label, value, type);
+                addIDs(hm, label, value);
+                addStandardNumber(hm, label, value);
+
+                if ("FAU".equals(label)) {
                     if ("".equals(author)) {
-                        author = val;
+                        author = value;
                     } else {
-                        author += " and " + val;
+                        author += " and " + value;
                     }
-                } else if ("FED".equals(lab)) {
+                } else if ("FED".equals(label)) {
                     if ("".equals(editor)) {
-                        editor = val;
-                    } else {
-                        editor += " and " + val;
-                    }
-                } else if ("JT".equals(lab)) {
-                    if ("inproceedings".equals(type)) {
-                        hm.put("booktitle", val);
+                        editor = value;
                     } else {
-                        hm.put("journal", val);
+                        editor += " and " + value;
                     }
-                } else if ("PG".equals(lab)) {
-                    hm.put("pages", val);
-                } else if ("PL".equals(lab)) {
-                    hm.put("address", val);
-                } else if ("IS".equals(lab)) {
-                    hm.put("issn", val);
-                } else if ("VI".equals(lab)) {
-                    hm.put("volume", val);
-                } else if ("AB".equals(lab)) {
-                    String oldAb = hm.get("abstract");
-                    if (oldAb == null) {
-                        hm.put("abstract", val);
+                } else if ("PG".equals(label)) {
```

Comment:
Could you please store the conversation as a map<MedlineKey, Bibtexkey> and then iterate over this map. Should be more readable.